src/sentry/incidents/subscription_processor.py

```diff
@@ -500,6 +506,86 @@ def process_update(self, subscription_update: QuerySubscriptionUpdate) -> None:
 
         aggregation_value = self.get_aggregation_value(subscription_update)
 
+        self.has_feature_flag = features.has(
+            "organizations:anomaly-detection-alerts", self.subscription.project.organization
+        )
+
+        if self.has_feature_flag:
+            # MF: I would prefer not to have this all be in an if statement. But what can you do ¯\_(ツ)_/¯
```

Comment:
You could put this logic into a separate function - that actually might make this easier to test without being able to trigger metric alerts locally.